hs_file_types/views.py

```diff
@@ -481,12 +481,15 @@ def get_metadata(request, hs_file_type, file_type_id, metadata_mode):
     if json_response is not None:
         return json_response
 
-    if metadata_mode == 'view':
-        metadata = logical_file.metadata.get_html()
-    else:
-        metadata = logical_file.metadata.get_html_forms()
+    try:
+        if metadata_mode == 'view':
+            metadata = logical_file.metadata.get_html()
+        else:
+            metadata = logical_file.metadata.get_html_forms()
+        ajax_response_data = {'status': 'success', 'metadata': metadata}
+    except Exception as ex:
+        ajax_response_data = {'status': 'error', 'message': ex.message}
 
-    ajax_response_data = {'status': 'success', 'metadata': metadata}
     return JsonResponse(ajax_response_data, status=status.HTTP_200_OK)
```

Comment:
@pkdash It appears here that even if there is an exception that causes a status of error, the return of JsonResponse() still returns with a HTTP_200_OK status which means success. I guess it depends on how the ajax request responds to this response, but it's probably better and more reasonable locally to return a failed status in JsonResponse.